src/scripts/main.js

```diff
@@ -1,11 +1,31 @@
+/* eslint-disable prettier/prettier */
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  const promise = new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      // eslint-disable-next-line no-undef
+      resolve(`It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`);
+    },
+    { once: true });
+  });
+
+  return promise;
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const existingMessage = document.querySelector('.message');
```

Comment:
The variable 'existingMessage' is only used inside the 'printMessage' function. Consider declaring it inside the function to limit its scope and avoid potential naming conflicts.